### PR TITLE
Make sidebar CSS use semantic vars

### PR DESCRIPTION
**Pull request by cdegroot**
Closes https://github.com/elixir-lang/ex_doc/issues/1668

I decided to do a relatively clean implementation of the sidebar, keeping the color scheme with neutral greys plus the language accent colors. Most work was making all the CSS for the sidebar use semantically-named variables which I think is a good idea anyway. 

Disclaimer: I'm not a designer, which is mainly why I stuck with the existing color definitions. I do think that I created a reasonably faithful "inverse" of the current dark-mode-only sidebar, though. More than happy to see tweaks, of course.

Note: the highlighted item still doesn't pop out due to the too-thin rendering of Lato on Linux/X11. However, in this mode it only impacts how much it stands out, not the readability itself so it is less of an issue.

image

### CODE REVIEW

**Comment by josevalim** [comment]
Hi @cdegroot! This is a 1-by-1 translation, right? So this looks good to me. There is just one TODO that you added and we can ship it?

**Comment by cdegroot** [comment]
Is there a todo? I missed that then :). Will check later today and clean that up. And give it a review in daylight to see whether colors want tweaking.

**Comment by cdegroot** [comment]
Fixed todo. And yes, it's an 1-1 translation. I followed a two step process:
* Move all the colors in sidebar.css to semantically named variables in the dark mode theme css
* Copy the resulting list of variables over to the light mode theme css and tweak until things look right. 

I kept the "language accent colors" so Elixir and Erlang still have their purple/orange accents.

Even if this results in follow-up discussions/tweaks, it should now be much simpler to work on. 

Screenshot after the tweak of commit 17f989b

image

File: assets/css/custom-props/_erlang.css [hunk 1]
```diff
@@ -2,6 +2,7 @@
   --main:               hsl(0, 100%, 64%);
   --main-darkened-10:   hsl(0, 100%, 54%);
   --main-darkened-20:   hsl(0, 100%, 44%);
+  --main-darkened-50:   hsl(0, 100%, 14%);
```
Comment (josevalim) [inline assets/css/custom-props/_erlang.css:5]: This does not seem to be used anywhere.

File: assets/css/custom-props/theme-light.css [hunk 2]
```diff
@@ -59,6 +59,23 @@
 
   --success:                     var(--green);
 
-  --sidebarButton:               var(--gray50);
-  --sidebarButtonBackground:     linear-gradient(180deg, var(--white) 20%, var(--white-opacity-50) 70%, var(--white-opacity-0) 100%);
+  --sidebarButtonBackground:     linear-gradient(180deg,
+                                                 var(--gray900) 20%,
+                                                 var(--gray900-opacity-50) 70%,
+                                                 var(--gray900-opacity-0) 100%);
```
Comment (josevalim) [inline assets/css/custom-props/theme-light.css:65]: It seems you changed the color compared to the previous theme.

File: assets/css/sidebar.css [hunk 3]
```diff
@@ -442,15 +442,15 @@
...
+  color: var(--sidebarHover);
 }
 
 .sidebar-button:is(:active, :hover, :focus) {
   outline: none;
 }
 
 .sidebar-button {
-  color: var(--sidebarButton);
+  color: var(--sidebarAccentMain);
```
Comment (josevalim) [inline assets/css/sidebar.css:453]: This was `--gray50` for light mode but now it is `--gray800`.

**Review by josevalim — CHANGES_REQUESTED** [review]
Thank you for the PR. I did an initial pass and it seems the values are changing in some occasions. Can you please double check all vars to guarantee there is no change in behaviour?

**Comment by josevalim** [comment]
Sorry, I got confused. I thought this was only an extraction of the sidebar into variables for easier change by plugins. But it is a new sidebar color altogether. We don't plan to merge this, see https://github.com/elixir-lang/ex_doc/issues/1668.

**Comment by cdegroot** [comment]
So is it acceptable if I just push the first commit? That makes it easier to override styling using things like the Stylus browser plugin but has no visible changed.

**Comment by josevalim** [comment]
Yes, I would be glad to accept that, as long as no changes in behaviour!

**Pull request closed without merging**